lib/data/entity.js

```diff
@@ -86,8 +88,10 @@ const parseSubmissionXml = (entityFields, xml) => new Promise((resolve, reject)
 // request data in the form of JSON.
 const extractEntity = (body, propertyNames, existingEntity) => {
   const bodyKeys = ['label', 'uuid', 'data'];
-  if (Object.keys(body).filter(k => !bodyKeys.includes(k)).length > 0)
-    throw Problem.user.invalidEntity({ reason: 'Unrecognized fields included in request.' });
+
+  const unexpectedKeys = Object.keys(body).filter(k => !bodyKeys.includes(k));
+  if (unexpectedKeys.length > 0)
+    throw Problem.user.unexpectedAttributes({ fields: unexpectedKeys });
```

Comment:
Let's say you have top-level json properties named `age` and `fav_color` because you forgot to nest in `data`, the message would say `Passed parameters age, fav_color were not expected.`? That doesn't feel all that helpful to me. Maybe stating the **expected** json properties would help? Something like "Expected top-level JSON properties: <expected>. Got: <actual>."